crates/miden-tx/src/host/kernel_process.rs

```diff
@@ -149,55 +149,43 @@ impl<'a> TransactionKernelProcess for ProcessState<'a> {
         Ok((inputs, script_root, serial_num))
     }
 
-    /// Extracts and validates note inputs from the advice provider using trial unhashing.
-    ///
-    /// This function tries to determine the correct number of inputs by:
-    /// 1. Finding the last non-zero element as a starting point
-    /// 2. Building NoteInputs and checking if the hash matches inputs_commitment
-    /// 3. If not, incrementing num_inputs and trying again (up to 6 more times)
-    /// 4. If num_inputs grows to the size of inputs_data and there's still no match, returning an
-    ///    error
+    /// Extracts and validates note inputs from the advice provider.
     fn read_note_inputs_from_adv_map(
         &self,
         inputs_commitment: &Word,
     ) -> Result<NoteInputs, TransactionKernelError> {
         let inputs_data = self.advice_provider().get_mapped_values(inputs_commitment);
+        let inputs_commitment_hash = Hasher::hash_elements(inputs_commitment.as_elements());
+        let num_inputs = self.advice_provider().get_mapped_values(&inputs_commitment_hash);
```

Comment:
nit: why not do this inside the `match` statement (i.e., after line 163)? We only need these values if `inputs_data` is `Some`, right?

Reply:
Yes, updated!